test/src/uploadFile_test.js

```diff
@@ -95,4 +95,79 @@ describe('Upload file from', async () => {
       }),
     );
   });
+
+  context('remote upload cancellation', () => {
+    it(
+      'can be cancelled',
+      vcr(async () => {
+        let noProgress = true;
+        const promise = uploadFile(
+          client,
+          'https://www.datocms-assets.com/13095/1561723946-happyfoxbymazack-d8u2l0s-2.jpeg',
+          {},
+          {},
```

Comment:
To preserve backwards compatibility we need to pass these guys when we want to pass `onProgress`. This is unfortunate but if you think it makes sense I can explore the option to overload it i.e. make it work with the old signature and a new one:

```js
uploadFile(client, string | File, options)
```

With this new signature `options` will include the other options as well

Reply:
Never really liked the `uploadFile` method in the first place, as it tries to do too many things togheter (create upload request, upload to S3, create the upload, return a Single asset file compatible object).. I'm planning to remove it from the docs in favour of the more explicit `.createUploadPath` and `.uploads.create` methods, so don't worry about it! 👍